sky/cli.py

```diff
@@ -3004,7 +3004,7 @@ def _down_or_stop_clusters(
                     # with the termination.
                     hint_or_raise(controller_name, purge)
                 except (exceptions.ClusterOwnerIdentityMismatchError,
-                        RuntimeError) as e:
+                        exceptions.NotSupportedError, RuntimeError) as e:
```

Comment:
Preferrably, we may want to distinguish the following two cases:
1. The controller cannot be reached, i.e. either the controller is terminated externally, or the network issue
2. We actually detected that there are still jobs on the controller.

It is ok to purge the controller with the first case.

For the second case, we probably should not allow the termination the controller no matter if `purge` is specified and ask the user to run `sky jobs cancel -a` first. I met a few times with this by accidently have a `-y` in `sky down` controller.

Reply:
@Michaelvll I was testing this a bit yesterday, and `sky down -y` will not pass the "type `delete`" check. In the new smoke test for #4787, we do `echo delete | sky down`. Still, do you think we should avoid allowing this to be circumvented?

I guess there could be some cases where `sky jobs cancel` doesn't work, like a job is stuck in PENDING but cannot be scheduled for some reason. (Due to some bug or something.) Ideally the user can still intervene in this case. But maybe deleting from the cloud console is sufficient. What do you think?

Reply:
Ahh, if `-y` does not pass the deletion prompt, it should be fine to keep the current impl. Thanks for the explanation @cg505!